processor/src/main/kotlin/dev/zacsweers/autoservice/ksp/AutoServiceSymbolProcessor.kt

```diff
@@ -136,9 +141,29 @@ public class AutoServiceSymbolProcessor(environment: SymbolProcessorEnvironment)
           }
         }
     generateAndClearConfigFiles()
-    return emptyList()
+    return result.toList()
+  }
+
+  private fun isCorrectlyAnnotated(provider: KSClassDeclaration, autoServiceType: KSType): Boolean {
+      val annotation = provider.annotations.find { it.annotationType.resolve() == autoServiceType }
+      if (annotation == null) {
+          logger.error("@AutoService annotation not found", provider)
+          return false
+      }
+  
+      try {
+          val argumentValue = annotation.arguments.find { it.name?.getShortName() == "value" }?.value
+          val providerInterfaces = argumentValue as? List<KSType> ?: listOf(argumentValue as KSType)
+          return providerInterfaces.isNotEmpty() && providerInterfaces.all { interfaceType ->
+              provider.getAllSuperTypes().any { it.isAssignableFrom(interfaceType) }
```

Comment:
We do this validation already below, is this now duplicating that? I was thinking this could just be as simple as checking if the KSType is an error type and deferring the current symbol if so. Let's avoid duplicating work or using exceptional control flow like this.

For testing, it should be doable but you'll have to have two processors: one that generates a type that's used in an autoservice annotation and then this one after. Then make the test source use the generated type in its autoservice annotation or as a super type (or both!)